### ISSUE TITLE
[FR] Limit rollover to a specific heading

### ISSUE THREAD

**Issue by guybrusht** [lumoe/obsidian-rollover-daily-todos#68]
Amazing plugin! 
Just one thing that could be really helpful: I have tasks under a heading that I have specified in the settings. I would like that only the tasks under that particular heading be rolled over to the next day and not those of the whole note.

For instance, if my template looks like this:

> # Big header
> ## Something Something
> - [ ] Task 1
> - [ ] Task 2
> ## ToDo
> - [ ] Task 3
> - [ ] Task 4
> ## Something Else
> - [ ] Task 5
> Blabla
> # Big header 2
> - [ ] Task 6
> - [ ] Task 7
> 

Would it be possible to only rollover what is under the ToDo heading (e.g. Tasks 3 & 4) and nothing else?
Cheers

Probably related to #54

**Comment by reorganice** [comment]
Yes, its very nice to have this plugin come back to life. Thank you very much, @lumoe !

I would also love to have this
**possibility to specify a headline where todos originate from** feature
as I use checkboxes for checking my daily table of contents on top of every daily note.

<img width="434" alt="daily-directory" src="https://user-images.githubusercontent.com/103707774/220094739-19806085-569e-452d-a032-7ef4c0d663a7.png">

**Comment by tim-hilt** [comment]
This should already be possible with the release-2.0 branch. Can you to test if the approach works for you?

**Comment by reorganice** [comment]
Ok now I finally figured it out, how to use the 2.0 version :)
Would be nice if you change the version number in the package.json already?
This way the version beeing used would be shown in the Obsidian Settings Panel?

So I also get this error:
2023-02-20 is not a daily note.
files like this: vault -> !chronic/daily/2023-02-20.md

Furthermore:
I like the way I can choose one / severall Headings for where to take the TODOs from.
-> But I am missing the possibility to Rollover these TODOs to a specific Heading only,
like it used to be in the old version.

Thank you in advance, Tim !

**Comment by tim-hilt** [comment]
> So I also get this error:
2023-02-20 is not a daily note.
files like this: vault -> !chronic/daily/2023-02-20.md

can you provide a screenshot of the settings for your daily-notes plugin and rollover-daily-todos, as well as the full path of `2023-02-20.md` inside your vault? This would help me with debugging :)

As for the second part: The old behavior of collecting all todos under one specific heading isn't planned to further exist. It conflicts with the multi-heading approach. Maybe it will become a configuration-option if enough users request this feature to further exist.

**Comment by reorganice** [comment]
full path is this:
vault folder name: _interconnect -> _interconnect/!chronic/daily/2023-02-20.md

<img width="1060" alt="settings-daily" src="https://user-images.githubusercontent.com/103707774/220140520-b8cb9dd5-91b6-462d-bb70-edb26ac8fd22.png">
<img width="822" alt="settings-rollover2" src="https://user-images.githubusercontent.com/103707774/220140526-b9ad21d1-ac17-44c5-b548-885630ba5518.png">

I would like them rollovered checkboxes to come under
# rest erledigt ? (rest done?) 
I really prefer this way, instead of polluting my new todos :)
Nice would be to have the possibility to have them wrapped inside a callout as well...

Now I see this callout stuff might have killed your approach...

<img width="417" alt="todos-section" src="https://user-images.githubusercontent.com/103707774/220140531-08ca2cf0-f60d-46f6-ac87-24dc2d8eb64b.png">

No stress, brother :D

**Comment by tim-hilt** [comment]
Your issue lead to me finding not one, but two bugs! :D Can you check if it works for you now?

**Comment by reorganice** [comment]
So, the rollover is working !

I get still get this error...
<img width="333" alt="rollover-error2" src="https://user-images.githubusercontent.com/103707774/220882029-49430152-8084-4b33-8d1f-0c952ec3de14.png">

It works but:
Everything (not the specified heading only) gets rollovered 
to the end of the destination note.

Extra information for you:
When I look into the Plugin settings, i see that the source headings are not up to date.
I changed some in the meantime (not the source heading) but it won't reflect in the settings at any point. 

My Case (just wishing, might not be needed):
It would be nice to take my _callout headings_ and my _## rest erledigt_ inside my main _# Todos :_ with me. Maybe even a blank line seperating my different sections would do the job, too.
My conclusion for me:
I might change my Todos section accordingly. Insert checkbox headings for the different sections.
But in the end, for my approach, it should be possible to choose a / severall source headings 
and it would be nice to get them spilled out under one spicific heading (in my case: _## rest erledigt_).

I see this conflicts with your approach. But as described above I cannot make changes to check out your approach.

Thank you and well done for now!

**Comment by tim-hilt** [comment]
Thanks for checking it! Yes, refreshing the template-headings is not done at the moment. We basically assume, that your daily-notes-template is static and unchanged. I might have to come up with a cache-invalidation-strategy at some point, but I consider this to be out of scope for now.

To fix it for you: Uninstall and reinstall the plugin and activate the headings to your liking.

I'll leave this issue open though! It's a request that's not totally open from my POV. Thanks again for checking it and helping me find those two bugs!

**Comment by reorganice** [comment]
Wow, allright!

I deleted all that callout stuff & put everything on lowest level.
Now, stuff gets copied / moved from & to the appropriate headline.
Everything working as aspected.

Thanks a lot !! ✨🥳

**Comment by kxc0re** [comment]
how easy can you integrate a button to refresh the template headings manually?
my old template seems to stay cached even after reinstalling the release-2 branch

**Comment by tim-hilt** [comment]
@kxc0re that’s a good idea. I have no idea how Obsidian caches settings, but I noticed this problem as well.

**Comment by Mike-Ki** [comment]
Can someone briefly explain how I can set up Release 2.0 to only carry over tasks to the next day under a specific heading?

**Comment by tim-hilt** [comment]
See https://github.com/lumoe/obsidian-rollover-daily-todos/issues/80 for related discussion
